Sources/ClientRuntime/Telemetry/Context/TelemetryScope.swift

```diff
@@ -0,0 +1,12 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/// Delineates a Telemetry Scope that has a beginning and end, particularly for Telemetry Contexts.
```

Comment:
Q: "[P]articularly for"? Does `TelemetryScope` get used for other things that's not `TelemetryContext`?

Reply:
Given the current architecture of a Swift client's middleware stack, `TelemetryContext` is provided specifically for telemetry.

However, ideally this would be a "context" (which would at minimum act as a property bag) which is not specific to the telemetry. This doesn't exist in the Swift SDK (middleware context would be the closest).